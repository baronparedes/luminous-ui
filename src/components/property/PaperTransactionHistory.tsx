import React from 'react';

import {PrintPaper} from '../@print-papers/PaperPdf';
import TransactionHistory from '../@print-papers/TransactionHistory';

type Props = React.ComponentProps<typeof TransactionHistory>;

const PaperTransactionHistory = React.forwardRef<HTMLDivElement, Props>(
  ({...soaProps}, ref) => {
    return (
      <div className="d-none">
        <PrintPaper ref={ref}>
          <TransactionHistory {...soaProps} />
        </PrintPaper>
      </div>
    );
  }
);

export default PaperTransactionHistory;
