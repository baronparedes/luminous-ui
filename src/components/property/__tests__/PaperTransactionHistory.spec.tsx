import {render} from '@testing-library/react';

import {ChargeAttr} from '../../../Api';
import PaperTransactionHistory from '../PaperTransactionHistory';

describe('PaperTransactionHistory', () => {
  const charge: ChargeAttr = {
    code: 'ESTATE DUES',
    chargeType: 'unit',
    communityId: 1,
    postingType: 'monthly',
    rate: 100,
  };
  const props: React.ComponentProps<typeof PaperTransactionHistory> = {
    property: {
      id: 1,
      code: 'G-111',
      address: 'Pasig City Philippines',
      floorArea: 33.1,
      status: 'active',
    },
    data: {
      targetYear: 2022,
      previousBalance: 0,
      transactionHistory: [
        {
          amount: 1000,
          charge: charge,
          chargeId: 1,
          propertyId: 1,
          transactionPeriod: '2022-01-01',
          transactionType: 'charged',
        },
        {
          amount: 1000,
          charge: charge,
          chargeId: 1,
          propertyId: 1,
          transactionPeriod: '2022-01-01',
          transactionType: 'collected',
        },
      ],
      paymentHistory: [
        {
          amount: 1000,
          code: 'Charge 1',
          collectedBy: 'Jhon Doe',
          orNumber: 'OR#1',
          paymentType: 'cash',
          transactionPeriod: '2022-01-01',
        },
        {
          amount: 1000,
          code: 'Charge 2',
          collectedBy: 'Jhon Doe',
          orNumber: 'OR#2',
          paymentType: 'check',
          transactionPeriod: '2022-02-01',
          checkIssuingBank: 'Bank of America',
          checkNumber: 'BA-123-123',
          checkPostingDate: '2022-02-28',
        },
      ],
    },
  };

  it('should render', () => {
    const {asFragment} = render(<PaperTransactionHistory {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
