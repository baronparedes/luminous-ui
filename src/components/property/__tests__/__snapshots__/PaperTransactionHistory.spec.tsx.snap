// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaperTransactionHistory should render 1`] = `
<DocumentFragment>
  <div
    class="d-none"
  >
    <div>
      <div
        class="sc-bdfBQB dxnWgn"
      >
        <div
          class="sc-bdfBQB dxnWgn"
        >
          <div
            class="text-center"
          >
            <small>
              <h4
                class="brand"
              >
                Luminous
              </h4>
            </small>
            <h4>
              TRANSACTION HISTORY
            </h4>
          </div>
        </div>
        <div
          class="sc-bdfBQB dxnWgn pt-3"
        >
          <div
            class="sc-gsTEea bjzFuM"
          >
            <strong>
              Unit Number: 
            </strong>
            G-111
          </div>
          <small>
            <div
              class="sc-gsTEea bjzFuM"
            >
              <strong>
                Address: 
              </strong>
              Pasig City Philippines
            </div>
          </small>
          <small>
            <div
              class="sc-gsTEea bjzFuM"
            >
              <strong>
                Floor area: 
              </strong>
              33.1
            </div>
          </small>
          <hr />
        </div>
        <div
          class="sc-bdfBQB dxnWgn pt-3"
        >
          <div
            class="sc-gsTEea bjzFuM"
          >
            <strong>
              Transaction History for: 
            </strong>
            2022
          </div>
          <hr />
        </div>
        <div
          class="sc-bdfBQB dxnWgn"
        >
          <div
            class="text-right mb-3 container"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <div
                  class="text-center"
                >
                  <span
                    class="badge badge-pill badge-danger"
                  >
                    previous balance
                  </span>
                  <h4>
                    <span
                      class="currency"
                    >
                      0
                    </span>
                  </h4>
                </div>
              </div>
              <div
                class="col"
              >
                <div
                  class="text-center"
                >
                  <span
                    class="badge badge-pill badge-info"
                  >
                    running balance
                  </span>
                  <h4>
                    <span
                      class="currency"
                    >
                      0
                    </span>
                  </h4>
                </div>
              </div>
            </div>
          </div>
          <div
            class="list-group"
          >
            <div
              class="list-group-item"
            >
              <div
                class="p-0 m-0 container"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-md-2"
                  >
                    <div>
                      <h5>
                        JAN
                      </h5>
                    </div>
                  </div>
                  <div
                    class="col"
                  >
                    <div
                      class="list-group"
                    >
                      <div
                        class="p-2 list-group-item"
                      >
                        <small>
                          <div
                            class="d-inline pr-2"
                          >
                            <span
                              class="text-muted pr-2"
                            >
                              Charged
                            </span>
                            <strong>
                              ESTATE DUES
                            </strong>
                          </div>
                          <div
                            class="d-inline pr-2"
                          >
                            <span
                              class="text-muted pr-2"
                            >
                              with an amount of
                            </span>
                            <strong>
                              1000
                            </strong>
                          </div>
                        </small>
                      </div>
                      <div
                        class="p-2 list-group-item"
                      >
                        <small>
                          <div
                            class="d-inline pr-2"
                          >
                            <strong>
                              <span
                                class="text-muted pr-2"
                              >
                                OR#
                              </span>
                              OR#1
                            </strong>
                          </div>
                          <div
                            class="d-inline pr-2"
                          >
                            <span
                              class="text-muted pr-2"
                            >
                              received
                            </span>
                            <strong>
                              cash
                            </strong>
                          </div>
                          <div
                            class="d-inline pr-2"
                          >
                            <span
                              class="text-muted pr-2"
                            >
                              with an amount of
                            </span>
                            <strong>
                              <span
                                class="currency text-success"
                              >
                                1,000
                              </span>
                            </strong>
                          </div>
                        </small>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="list-group-item"
            >
              <div
                class="p-0 m-0 container"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-md-2"
                  >
                    <div>
                      <h5>
                        FEB
                      </h5>
                    </div>
                  </div>
                  <div
                    class="col"
                  >
                    <div
                      class="list-group"
                    >
                      <div
                        class="p-2 list-group-item"
                      >
                        <small>
                          <div
                            class="d-inline pr-2"
                          >
                            <strong>
                              <span
                                class="text-muted pr-2"
                              >
                                OR#
                              </span>
                              OR#2
                            </strong>
                          </div>
                          <div
                            class="d-inline pr-2"
                          >
                            <span
                              class="text-muted pr-2"
                            >
                              received
                            </span>
                            <strong>
                              check
                            </strong>
                          </div>
                          <div
                            class="d-inline pr-2"
                          >
                            <span
                              class="text-muted pr-2"
                            >
                              with an amount of
                            </span>
                            <strong>
                              <span
                                class="currency text-success"
                              >
                                1,000
                              </span>
                            </strong>
                          </div>
                          <div
                            class="text-muted"
                          >
                            <small
                              class="pr-3"
                            >
                              <span>
                                Bank of America
                              </span>
                            </small>
                            <small
                              class="pr-3"
                            >
                              <span>
                                BA-123-123
                              </span>
                            </small>
                            <small>
                              <span>
                                2022-02-28
                              </span>
                            </small>
                          </div>
                        </small>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
